compaction/task_manager_module.hh

```diff
@@ -466,6 +469,87 @@ protected:
     virtual future<> run() override;
 };
 
+class reshaping_compaction_task_impl : public compaction_task_impl {
+public:
+    reshaping_compaction_task_impl(tasks::task_manager::module_ptr module,
+            tasks::task_id id,
+            unsigned sequence_number,
+            std::string keyspace,
+            std::string table,
+            std::string entity,
+            tasks::task_id parent_id) noexcept
+        : compaction_task_impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(entity), parent_id)
+    {
+        // FIXME: add progress units
+    }
+
+    virtual std::string type() const override {
+        return "reshaping compaction";
+    }
+protected:
+    virtual future<> run() override = 0;
+};
+
+class reshaping_table_compaction_task_impl : public reshaping_compaction_task_impl {
+private:
```

Comment:
for symmetry with others, table_reshape_compaction_task instead? which also makes more sense anyway.